engines/director/debugger/dt-scripts.cpp

```diff
@@ -351,5 +294,223 @@ void showFuncList() {
 	ImGui::End();
 }
 
+void showExecutionContext() {
+	if (!_state->_w.executionContext)
+		return;
+
+	ImGui::SetNextWindowPos(ImVec2(20, 160), ImGuiCond_FirstUseEver);
+	ImGui::SetNextWindowSize(ImVec2(500, 750), ImGuiCond_FirstUseEver);
+
+	Director::Lingo *lingo = g_director->getLingo();
+
+	Window *currentWindow = g_director->getCurrentWindow();
+
+	if (ImGui::Begin("Execution Context", &_state->_w.executionContext, ImGuiWindowFlags_AlwaysAutoResize)) {
+		Window *stage = g_director->getStage();
+		g_director->setCurrentWindow(stage);
+		g_lingo->switchStateFromWindow();
+
+		int windowID = 0;
+		ImGui::PushID(windowID);
+		ImGui::PushStyleColor(ImGuiCol_ChildBg, ImGui::GetStyleColorVec4(ImGuiCol_FrameBg));
+
+		ImGui::BeginChild("Window##", ImVec2(500.0f, 700.0f));
+		ImGui::Text("%s", stage->asString().c_str());
+		ImGui::Text("%s", stage->getCurrentMovie()->getMacName().c_str());
+
+		ImGui::SeparatorText("Backtrace");
+		ImVec2 childSize = ImGui::GetContentRegionAvail();
+		childSize.y /= 3;
+		ImGui::BeginChild("##backtrace", childSize);
+		ImGui::Text("%s", lingo->formatCallStack(lingo->_state->pc).c_str());
+		ImGui::EndChild();
+
+		ImGui::SeparatorText("Scripts");
+
+		_state->_functions._scriptData = &_state->_functions._windowScriptData.getOrCreateVal(stage);
```

Comment:
Why do you need to keep a pointer to it like this? E.g. window ould be destroyed in the meantime and you get crash. Why not keep it as an key to the hashmap?